### DISCUSSION TITLE
Generator: 关于带形参函数的定义中使用形参的问题

### DISCUSSION THREAD

**Discussion by zhenglq23** [arcsysu/SYsU-lang#98] [Q&A]
对形参分配了空间，但函数中若有需要访问该形参的情况就会报错，比如代码：
```
int func(int p){
  p=10;
  return p;
}
```
报错如下：
`sysu-generator: /usr/lib/llvm-11/include/llvm/IR/Type.h:371: llvm::Type *llvm::Type::getPointerElementType() const: Assertion getTypeID() == PointerTyID' failed.`
但对于代码
```
int func(int p){
     p=10; 
     return 0;
}
```
得到运行结果如下：
```
define i32 @func(i32 %p) {
entry:
  %p1 = alloca i32, align 4
  store i32 10, i32 %p, align 4
  ret i32 0
}
```

请问有没有相关资料网页可以对此问题学习一下？

**Comment by zejia-lin** [comment]
应该是`store i32 10, i32 %p1, align 4`，函数形参并不持有内存地址，所以不能store到形参里
